ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/AutoCreateIcebergTagAction.java

```diff
@@ -109,22 +104,27 @@ private boolean exceedMaxDelay(Snapshot snapshot) {
     if (tagConfig.getMaxDelayMinutes() <= 0) {
       return false;
     }
-    long delay = snapshot.timestampMillis() - getTagTriggerTime();
+    long delay = snapshot.timestampMillis() - getTagTriggerTimestampMillis();
     return delay > tagConfig.getMaxDelayMinutes() * 60_000L;
   }
 
   private String generateTagName() {
-    if (tagConfig.getTriggerPeriod() == TagConfiguration.Period.DAILY) {
-      String tagFormat = tagConfig.getTagFormat();
-      return now.minusDays(1).format(DateTimeFormatter.ofPattern(tagFormat));
-    } else {
-      throw new IllegalArgumentException(
-          "unsupported trigger period " + tagConfig.getTriggerPeriod());
-    }
+    LocalDateTime tagTime =
```

Comment:
Hi, when I reviewed the code here, these three time-related variables made it difficult for me to understand. After carefully reviewing the code again, I have summarized the names and meanings of the three parameters:

- Check time: the timestamp when the action is started.
- Trigger time: the timestamp of the last tag should be triggered.
- Tag time: the timestamp of the last tag represents.

And we calculate the 3 variables with the methods:

- checkTime = now()
- triggerTime = (checkTime - triggerOffset) % period + triggerOffest
- TagTime = triggerTime - triggerOffest - period

then,
-  TagTime = (checkTime - triggerOffset) % period - period
- TriggerTime = TagTime + period + triggerOffset

And I think we can make the code much easier to understand by:

- Put the 3 variables together and try to calculate them at the beginning, put some comments to tell the meaning and the calculating rules.
- Calculate the tag time first and then calculate the trigger time based on tag time.

Of course, the second point is just my personal suggestion, as this calculation process is easier for me to understand.